### PR TITLE
add 1-Two-Sum

### PR DESCRIPTION
**Pull request by Rwatana**
Pleasecheck my code

### CODE REVIEW

File: 1_two_sum.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from typing import List
+
+
+class Solution:
+    def twoSum(self, nums: List[int], target: int) -> List[int]:
+        lookup = {}
```
Comment (ryujumoto) [inline 1_two_sum.py]: You should name the dict variable "from_xxx_to_yyy", so in this case it's better to use "from_num_to_supplement" for example.

**Review by ryujumoto — COMMENTED** [review]
Done.

**Comment by Rwatana** [comment]
I changed the name from lookup. Please check

File: 1_two_sum.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+
+class Solution:
+    def twoSum(self, nums: List[int], target: int) -> List[int]:
+        from_num_to_supplement = {}
+        for i, num in enumerate(nums):
+            if target - num in from_num_to_supplement:
+                return [from_num_to_supplement[target - num], i]
+            from_num_to_supplement[num] = i
+            return []
```
Comment (ryujumoto) [inline 1_two_sum.py]:
```
for i, num in enumerate(nums):
    if target - num in from_num_to_supplement:
        return [from_num_to_supplement[target - num], i]

    from_num_to_supplement[num] = i

return []
```

File: 1_two_sum.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# time: O(n)
+# space: O(1)
```
Comment (ryujumoto) [inline 1_two_sum.py]: The space complexity is O(n).

**Comment by Rwatana** [comment]
I changed.
Please check my code.

**Comment by ryujumoto** [comment]
The part I pointed out has not been corrected.

**Comment by Rwatana** [comment]
I made the blank between between 11 and 12 // 12 and 13. I changed the word from "time: O(1)" to "O(n) " .
Is there any points that `I have to revise in my code.

File: 1_two_sum.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    def twoSum(self, nums: List[int], target: int) -> List[int]:
+        from_num_to_supplement = {}
+        
+        for i, num in enumerate(nums):
+            if target - num in from_num_to_supplement:
+                return [from_num_to_supplement[target - num], i]
+            
+            from_num_to_supplement[num] = i
+            
+            return []
```
Comment (ryujumoto) [inline 1_two_sum.py]: This return statement has to be outside of the for loop.

**Review by ryujumoto — COMMENTED** [review]
Check my review.

**Comment by Rwatana** [comment]
Thanks. I revised it.

**Review by ryujumoto — APPROVED** [review]

**Pull request merged**